idaes/models_extra/column_models/tests/test_utilities.py

```diff
@@ -0,0 +1,330 @@
+"""
+Tests for column model utitliy functions
+
+Author: Alejandro Garciadiego, Andrew Lee
+"""
+import pytest
+from types import MethodType
+
+from pyomo.environ import ConcreteModel, Block, Var, Expression, Reference, Set, value
+from pyomo.network import Port
+
+from idaes.models_extra.column_models.util import make_phase_split
+
+# Mark module as all unit tests
+pytestmark = pytest.mark.unit
+
+
+@pytest.fixture
+def m():
+    # Mock up something that looks like a tray model
+    m = ConcreteModel()
+    m.time = Set(initialize=[1, 2, 3])
+
+    def flowsheet(m):
+        return m
+
+    m.flowsheet = MethodType(flowsheet, m)
+
+    m.properties_out = Block(m.time)
+
+    m.port = Port()
+
+    return m
+
+
+def test_make_phase_split_general(m):
+    # Test for variables without special indices
+    # Mock define_port_members method
+    def define_port_members(m):
+        return {"temperature": m.temperature, "indexed": m.indexed}
+
+    # Add variables and define_port_members method to dummy state block
+    for i in m.time:
+        m.properties_out[i].temperature = Var()
+        m.properties_out[i].indexed = Var(["a", "b"])
+
+        m.properties_out[i].define_port_members = MethodType(
+            define_port_members, m.properties_out[i]
+        )
+
+    # Call make_phase_split function
+    make_phase_split(m, m.port)
+
+    # Check for expected results
+    for i in m.time:
+        assert m.port.temperature[i] is m.properties_out[i].temperature
+        for j in ["a", "b"]:
+            assert m.port.indexed[i, j] is m.properties_out[i].indexed[j]
+
+
+def test_make_phase_split_flow(m):
+    # Mock define_port_members method
+    def define_port_members(m):
+        return {
+            "flow_mol_phase": m.flow_mol_phase,
+            "flow_mol_phase_comp": m.flow_mol_phase_comp,
+        }
+
+    def _liquid_set(m):
+        return ["a1", "b1"]
+
+    def _vapor_set(m):
+        return ["a1", "b1"]
+
+    m.components = Set(initialize=["a1", "b1"])
+    m.phases = Set(initialize=["L", "V"])
+
+    m.config.property_package = DummyData()
+    m.config.property_package.component_list = m.components
+    m.config.property_package.phase_list = m.phases
+
+    m._liquid_set = MethodType(_liquid_set, m)
+
+    m._vapor_set = MethodType(_vapor_set, m)
+
+    # Add variables and define_port_members method to dummy state block
+    for i in m.time:
+        m.properties_out[i].flow_mol_phase = Var(m.phases)
+        m.properties_out[i].flow_mol_phase_comp = Var(m.phases, m.components)
+
+        m.properties_out[i].define_port_members = MethodType(
+            define_port_members, m.properties_out[i]
+        )
+
+    # Call make_phase_split function
+    make_phase_split(m, m.port)
+
+    # Check for expected results
+    for i in m.time:
+        for p in m.config.property_package.phase_list:
+            for j in m.config.property_package.component_list:
+                assert m.port.flow_mol_phase[i, p, j].is_expression_type()
```

Comment:
This doesn't really check that it is the expression you expect however. That said, checking the form of an expression can be difficult as the only way I know to really do it is to check the `str(expr.body)`.

You basically have two options - hard code the expected string (which is fragile as things might change in Pyomo that affect this), or build the expected expression manually and then compare to what you got from the method (which can be a lot of work for complex expressions).